### PR TITLE
Issue #533: rescue StandardError only

### PR DESCRIPTION
**Pull request by ColinDKelley**
Address issue #533 by only rescuing `StandardError` (the default) rather than `Exception`.

### CODE REVIEW

**Comment by ColinDKelley** [comment]
@ioquatix can you please review?
This is what you originally recommended: simply rescue `StandardError`. At the time I argued to rescue more, because Ruby could wind up burying other exceptions. But I'm backing away from that based on
1) Noise that was getting logged as reported here: https://github.com/guard/listen/pull/532
2) The fact that modern Ruby has `report_on_exception = true` by default, so exceptions that sneak by will still be logged to (to STDERR I believe, which is good enough in k8s). Also, such exceptions get re-raised if you `join` the thread later. These seem sufficient to address my concerns.

File: spec/lib/listen/adapter/linux_spec.rb [hunk 1]
```diff
@@ -12,34 +12,44 @@
   end
 
   if linux?
+  describe 'instance methods' do
```
[the remark this answers is not in the document]
Comment (ColinDKelley) [reply spec/lib/listen/adapter/linux_spec.rb:15]: intentional, to keep the PR from having noise from indentation for now

**Review by ioquatix — APPROVED** [review]

**Comment by ioquatix** [comment]
When rescuing `StandardError` I normally call the variable `error` rather than `exception`.

LGTM.

**Comment by ColinDKelley** [comment]
@ioquatix Re `error` vs. `exception`, I decided to leave it `exception` since all the surrounding methods call it that too. I don't see any harm in being a bit more precise. (All exceptions are errors, but not all errors are exceptions.)

**Pull request merged**